src/main/java/io/jenkins/plugins/ml/EditCodeAction.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2020 Loghi Perinpanayagam.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package io.jenkins.plugins.ml;
+
+import com.google.common.collect.ImmutableList;
+import hudson.FilePath;
+import hudson.model.*;
+import jenkins.model.ParameterizedJobMixIn;
+import jenkins.model.RunAction2;
+import jenkins.scm.api.SCMRevisionAction;
+import net.sf.json.JSONObject;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.DoNotUse;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.verb.POST;
+
+import javax.annotation.CheckForNull;
+import javax.servlet.ServletException;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class EditCodeAction implements RunAction2 {
```

Comment:
@Loghijiaha is this code copied from another plugin?

Reply:
I used https://www.jenkins.io/doc/developer/tutorial/extend/ this tutorial and got some related code form here https://github.com/svanoort/workflow-cps-plugin/blob/25bb2f5be7ee02264aeb374d6f8d708287ed4c2b/src/main/java/org/jenkinsci/plugins/workflow/cps/replay/ReplayAction.java